src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -1514,6 +1517,108 @@ def handle_assign_role_submission_event(
+
+    contact_plugin = plugin_service.get_active_instance(
+        db_session=db_session, project_id=incident.project.id, plugin_type="contact"
+    )
+    if not contact_plugin:
+        raise CommandError(
+            "Contact plugin is not enabled. Unable to list participants.",
+        )
+
+    active_participants = [p for p in participants if p.active_roles]
+    participant_list = []
+
+    for participant in active_participants:
+        participant_email = participant.individual.email
+        participant_info = contact_plugin.instance.get(participant_email, db_session=db_session)
+        participant_name = participant_info.get("fullname", participant.individual.email)
+        participant_list.append({"text": participant_name, "value": participant_email})
+
+    blocks = [
+        static_select_block(
+            label="Participant",
+            block_id=CreateTaskBlockIds.participant_select,
+            placeholder="Select Participant",
+            options=participant_list,
+        ),
+        Input(
+            label="Task Description",
+            element=PlainTextInput(
+                placeholder="Task description", initial_value="description", multiline=True
+            ),
+            block_id=CreateTaskBlockIds.description,
+        ),
+    ]
+
+    modal = Modal(
+        title="Create Task",
+        blocks=blocks,
+        submit="Create",
+        close="Close",
+        callback_id=CreateTaskActionIds.submit,
+        private_metadata=context["subject"].json(),
+    ).build()
+    client.views_open(trigger_id=body["trigger_id"], view=modal)
+
+
+def ack_create_task_submission_event(ack: Ack) -> None:
+    """Handles task creation acknowledgment."""
+    modal = Modal(
+        title="Create Task", close="Close", blocks=[Section(text="Creating task...")]
+    ).build()
+    ack(response_action="update", view=modal)
+
+
+@app.view(
+    CreateTaskActionIds.submit,
+    middleware=[action_context_middleware, db_middleware, user_middleware, modal_submit_middleware],
+)
+def handle_create_task_submission_event(
+    ack: Ack,
+    body: dict,
+    client: WebClient,
+    context: BoltContext,
+    db_session: Session,
+    form_data: dict,
+    user: DispatchUser,
+) -> None:
+    """Handles the create task submission."""
+    ack()
+
+    participant_email = form_data.get(CreateTaskBlockIds.participant_select).get("value", "")
+    owner = participant_service.get_by_incident_id_and_email(
+        db_session=db_session, incident_id=context["subject"].id, email=participant_email
+    )
+    incident = incident_service.get(db_session=db_session, incident_id=context["subject"].id)
+
+    task_in = TaskCreate(
+        assignees=[ParticipantUpdate.from_orm(owner)],
+        creator={"individual": {"email": user.email}},
+        description=form_data.get(CreateTaskBlockIds.description, ""),
+        incident=IncidentRead.from_orm(incident),
+    )
+    task_service.create(db_session=db_session, task_in=task_in)
```

Comment:
suggestion: i believe there's some opportunity to improve the UX at this stage of the flow and I'd recommend we create some connections here. 

after the task is created, where does it go? how do I find it in the UI? is there a direct link to it? how can i see my current tasks or all tasks for the incident.

i'm worried right now the tasks will get created and then become invisible basically.

Reply:
I also had similar feedback that after submitting, it would be great to have some kind of confirmation. Maybe even let the assignee know in the channel that the task was created? (As an ephemeral message or not.)

Reply:
Added a message to the channel when a task is created.  This message also includes a button that allows the users to resolve/reopen the task.
<img width="608" alt="Screenshot 2024-09-30 at 3 56 23 PM" src="https://github.com/user-attachments/assets/1d344ef1-1080-44a5-9ed5-ee8b0f0074dc">